File: Ejercicios/Java/Interfaces/EmailNotificador.java
interface INotificable {
    void notificar(String mensaje);
}

public class EmailNotificador implements INotificable {
    private String direccion;

    public EmailNotificador() {
        this("desconocido@mail.com");
    }

    public EmailNotificador(String direccion) {
        this.direccion = direccion;
    }

    public EmailNotificador(String direccion, String nombre) {
        this.direccion = nombre + " <" + direccion + ">";
    }

    public EmailNotificador(EmailNotificador otro) {
        this.direccion = otro.direccion;
    }

    @Override
    public void notificar(String mensaje) {
        System.out.println("Enviando '" + mensaje + "' a " + direccion);
    }
}

File: Ejercicios/JavaScript/Interfaces/EmailNotificador.js
class EmailNotificador {
  constructor(direccion, nombre) {
    if (direccion === undefined && nombre === undefined) {
      this.direccion = "desconocido@mail.com";
    } else if (direccion instanceof EmailNotificador) {
      this.direccion = direccion.direccion;
    } else if (nombre === undefined) {
      this.direccion = direccion;
    } else {
      this.direccion = `${nombre} <${direccion}>`;
    }
  }

  notificar(mensaje) {
    console.log(`Enviando '${mensaje}' a ${this.direccion}`);
  }
}

export default EmailNotificador;
